examples/HelloWorldExampleDS/HelloWorldPublisher.cpp

```diff
@@ -133,61 +146,62 @@ bool HelloWorldPublisher::init(Locator_t server_address)
 
 HelloWorldPublisher::~HelloWorldPublisher()
 {
-    Domain::removeParticipant(mp_participant);
+    DomainParticipantFactory::get_instance()->delete_participant(mp_participant);
 }
 
-void HelloWorldPublisher::PubListener::onPublicationMatched(
-    Publisher* /*pub*/,
-    MatchingInfo& info)
+void HelloWorldPublisher::PubListener::on_publication_matched(
+        DataWriter* /*pub*/,
+        const PublicationMatchedStatus& info)
 {
-    if (info.status == MATCHED_MATCHING)
+    if (info.current_count_change == 1)
     {
-        n_matched++;
```

Comment:
`n_matched` was an internal and application specific count of the matched endpoints. Right now you are instead saving `info.total_count`. As per documentation, the total count does not decrease if an endpoint leaves the communication. I think `current_count` should be used instead, or keep the count in the application.